ocs_ci/templates/workloads/cephfs_stress/cephfs_filesystem_monitor.py

```
@@ -0,0 +1,633 @@
+        
+        # Wait for threads to finish
+        for t in threads:
+            t.join(timeout=30)
+            if t.is_alive():
+                logger.warning(f"Thread {t.name} did not stop gracefully")
+        
+        logger.info("=" * 80)
+        logger.info("CephFS Filesystem Monitor Stopped")
+        if hang_detected:
+            logger.critical("HANG WAS DETECTED DURING MONITORING - Check hang marker files")
+            return 2
```

Ruff (0.15.12)

[warning] `return` inside `finally` blocks cause exceptions to be silenced

(B012)